src/hooks/attachments/useAttachment.ts

```diff
@@ -223,12 +223,14 @@ export default function useAttachment(
   }, [value])
 
   const canDownload = React.useMemo(() => {
+    const isDownloadableAttachment = (attachment: Attachment) =>
+      (attachment.type === 'NEW' ||
+        attachment.type === 'SAVING' ||
+        !attachment.type) &&
+      (!attachment.isPrivate || authService.isLoggedIn())
```

Comment:
Should be using the `useAuth()` hook here. As it will trigger changes, `authService.isLoggedIn()` will only be evaluated once.